tests/test_workflow_management.py

```diff
@@ -0,0 +1,204 @@
+import threading
+import uuid
+from typing import Callable
+
+import pytest
+
+# Public API
+from dbos import DBOS, Queue, SetWorkflowID
+from dbos._dbos import DBOSConfiguredInstance
+from dbos._error import DBOSWorkflowCancelledError
+from tests.conftest import queue_entries_are_cleaned_up
+
+
+def test_cancel_resume(dbos: DBOS) -> None:
+    steps_completed = 0
+    workflow_event = threading.Event()
+    main_thread_event = threading.Event()
+    input = 5
+
+    @DBOS.step()
+    def step_one() -> None:
+        nonlocal steps_completed
+        steps_completed += 1
+
+    @DBOS.step()
+    def step_two() -> None:
+        nonlocal steps_completed
+        steps_completed += 1
+
+    @DBOS.workflow()
+    def simple_workflow(x: int) -> int:
+        step_one()
+        main_thread_event.set()
+        workflow_event.wait()
+        step_two()
+        return x
+
+    # Start the workflow and cancel it.
+    # Verify it stops after step one but before step two
+    wfid = str(uuid.uuid4())
+    with SetWorkflowID(wfid):
+        handle = DBOS.start_workflow(simple_workflow, input)
+    main_thread_event.wait()
+    DBOS.cancel_workflow(wfid)
+    workflow_event.set()
```

Comment:
doesn't this `set` let the workflow continue to step 2, potentially before you `get_result`?

Reply:
No, as the workflow is cancelled and cannot execute another step (hence the next line)